src/main/java/core/basesyntax/Car.java

```diff
@@ -1,5 +1,27 @@
 package core.basesyntax;
 
-public class Car {
+public class Car implements Cloneable {
+    private String brand;
+    private String model;
+    private String version;
+    private int numberOfDoors;
+    private Engine engine;
 
+    public Car(String brand, String model, String version, int numberOfDoors, Engine engine) {
+        this.brand = brand;
+        this.model = model;
+        this.version = version;
+        this.numberOfDoors = numberOfDoors;
+        this.engine = engine;
+    }
+
+    @Override
+    public Car clone() throws CloneNotSupportedException {
+        brand = brand;
+        model = model;
+        version = version;
+        numberOfDoors = numberOfDoors;
+        engine = engine;
+        return new Car(brand, model, version, numberOfDoors, engine);
```

Comment:
you do not clone engine here
try creating a car, then clone it with `car.clone()`
and then try changing engine type on the original car

the engine type of the cloned car should not change